src/main/java/to/us/resume_builder/spring/controller/PDFController.java

```diff
@@ -10,43 +10,82 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import to.us.resume_builder.MiscUtils;
 import to.us.resume_builder.ResumeExporter;
 
+/**
+ * PDFController contains the /pdf endpoint allowing a caller to give a
+ * LaTeX string and compile it to PDF and have it returned directly or via
+ * file.io
+ *
+ * @author Jacob Curtis
+ * @author Micah Schiewe
+ */
 @RestController
 public class PDFController extends BasicController {
 	private static final Logger LOG = Logger.getLogger(PDFController.class.getName());
 
+	/**
+	 * This endpoint takes a latex string and a url boolean specifying if the
+	 * caller wants the compiled PDF file to be uploaded to file.io instead of
+	 * being directly returned in the response.
+	 *
+	 * @param latex The LaTeX string to be compiled to PDF.
+	 * @param url Whether the caller wants the PDF uploaded to file.io
+	 * @return The response the user requested, or possibly an error.
+	 */
 	@PostMapping("/pdf")
-	public ResponseEntity<InputStreamResource> postPdf(@RequestParam(name = "latex") String latex, @RequestParam(name = "url") boolean url) {
-		LOG.info(latex);
-		Path pdf = Path.of("./", "resume.pdf");
+	public ResponseEntity<InputStreamResource> postPdf(@RequestParam(name = "latex") String latex,
+													   @RequestParam(name = "url") boolean url) {
+		Path pdf;
+		do {
+			pdf = Path.of("./", MiscUtils.randomAlphanumericString(16) + ".pdf");
+		} while (Files.exists(pdf));
+
 		InputStreamResource rsc;
 		MediaType mediaType;
+		InputStream in;
 		try {
 			if (!ResumeExporter.export(pdf, latex)) {
-				LOG.warning("Export failed");
-				return getInternalServerErrorResponse("Export failed").build();
+				LOG.warning("PDF compilation failed");
+				return badRequest("Could not convert LaTeX to PDF").build();
 			}
+
 			LOG.info("requesting url: " + url);
 			if (url) {
+				// Upload PDF to file.io, return response
 				LOG.info("Uploading pdf to file.io");
 				String response = ResumeExporter.uploadPDF(pdf);
-				rsc = new InputStreamResource(new ByteArrayInputStream(response.getBytes()));
+				in = new ByteArrayInputStream(response.getBytes());
 				mediaType = MediaType.APPLICATION_JSON;
 			} else {
-				rsc = new InputStreamResource(new BufferedInputStream(new FileInputStream(String.valueOf(pdf))));
+				// Give the PDF in the response
+				in = new ByteArrayInputStream(String.valueOf(pdf).getBytes());
```

Comment:
I looked at the Javadocs for valueOf, and it looks like it just calls toString on the object, which in this case is just the filepath of the pdf and not the pdf's bytes itself. We'll need stream them in somehow, maybe by:
```java
try {
	byte[] file = Files.readAllBytes(pdf);
	in = new ByteArrayInputStream(file);
} catch (IOException ex) {
	LOG.logp(Level.WARNING, getClass().getName(), "postPdf", "IOException", ex);
}
```
We would need to be careful about OutOfMemoryExceptions, though--this operation is intended only for small files (Javadocs says smaller than 2GB).

Reply:
Fixed the part where it was just getting the filename. Was using a `FileInputStream` before and forgot to update that part of the code.

I don't think we really need to worry about `OutOfMemoryError`. Currently the PDF that gets compiled in our example is 32 KB.